### PR TITLE
Dev optimize branch

### PR DESCRIPTION
**Pull request by jiangjianyong**

### CODE REVIEW

File: src/TraceLaravel/TracingEventHandler.php [hunk 1]
```diff
@@ -159,14 +162,15 @@ protected function queryHandler($query, $bindings, $time, $connectionName)
...
-            'query.bindings' => $this->sqlBindings ? json_encode($query->bindings) : '******',
-            'query.time' => $query->time
-        ]);
+        $child = $this->service->newChild(
+            $query->connectionName . ':' . $query->sql,
+            [
+                'query.connectionName' => $query->connectionName,
+                'query.database' => $query->connection->getDatabaseName(),
+                'query.sql' => $query->sql,
+                'query.bindings' => $this->sqlBindings ? json_encode($query->bindings) : '******',
```
Comment (jcchavezs) [inline src/TraceLaravel/TracingEventHandler.php]: I think the best is that you don't include this if `$this->sqlBindings` is false.

File: src/TraceLaravel/TracingServiceProvider.php [hunk 2]
```diff
@@ -57,8 +50,16 @@ public function register()
...
+        $this->app->singleton(static::$abstract, function () {
+            return new TracingService(
+                config(static::$abstract),
+                request(),
+                new RequestHeaders()
+            );
+        });
+
+        $this->app->terminating(function () {
+            app(static::$abstract)->flush();
```
Comment (jcchavezs) [inline src/TraceLaravel/TracingServiceProvider.php]: Is this happening after the response is delivered to the user?

Comment (jiangjianyong) [reply src/TraceLaravel/TracingServiceProvider.php]: https://github.com/laravel/docs/blob/5.8/middleware.md#terminable-middleware

File: src/TraceLaravel/TracingEventHandler.php (hunk 1 above)
Comment (jiangjianyong) [reply src/TraceLaravel/TracingEventHandler.php]: Your suggestion is good

Comment (jiangjianyong) [reply src/TraceLaravel/TracingEventHandler.php]: https://github.com/laravelcloud/lar-trace/pull/8

**Pull request merged**